modifiers/hwloc/parse_hwloc_output.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2023 Lawrence Livermore National Security, LLC and other
+# Benchpark Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import re
+import json
+from collections import defaultdict
+import xmltodict
+
+
+def parse_hwloc_tree_full_metadata(obj, path="topology", results=None, counters=None):
+    if results is None:
+        results = []
+    if counters is None:
+        counters = defaultdict(int)
+
+    if isinstance(obj, dict):
+        raw_type = obj.get("@type", "Unknown")
+        counter = counters[raw_type]
+        counters[raw_type] += 1
+
+        type_display = f"{raw_type}[{counter}]"
+
+        node_info = {
+            "type": raw_type,
+            "instance": counter,
+        }
+
+        for key, value in obj.items():
+            if isinstance(value, (str, int, float)):
+                node_info[key] = value
+
+        if "info" in obj and isinstance(obj["info"], list):
+            for item in obj["info"]:
+                k = item.get("@name")
+                v = item.get("@value")
```

Comment:
`xmltodict` lib. by default prefix `@` to xml attributes when converting them from xml to json.